Documentation/In-depth/ComposerJson/Index.rst

```diff
@@ -0,0 +1,288 @@
+      },
+      "config": {
+         "platform": {
+            "php": "7.4.1"
+         },
+         "sort-packages": true
+      },
+      "extra": {
+         "typo3/cms": {
+            "web-dir": "public"
+         }
+      },
+      "scripts": {
+         "typo3-cms-scripts": [
+            "typo3cms install:generatepackagestates",
+            "typo3cms install:fixfolderstructure"
+         ],
+         "post-autoload-dump": [
+            "@typo3-cms-scripts"
+         ]
+      }
+   }
+
+You may notice that we include the third party extension
+`helhum/typo3-console` here. This is recommended and the
+section `scripts` uses it (see below).
+
+Commands for creating composer.json
+===================================
+
+It is possible to create a composer.json from scratch with
+`composer init`. However, in practice, it may be a good idea
+to have a general composer.json which you reuse for every project
+and add individual requirements for your projects via command line
+commands.
+
+In general it is recommended to use the command line to modify
+composer.json (as in `composer require ...`). If you modify
+a :file:`composer.json` in an existing project directly, you must
+execute a command like `composer install` or `composer require`
+afterwards to make sure that your :file:`composer.lock` will be updated.
+
+(Incomplete) example for creating above :file:`composer.json`
+(scripts is missing):
+
+.. code-block:: shell
+
+   composer init --type project; \
+   composer require ergebnis/composer-normalize; \
+   composer require helhum/typo3-console:^6.4.0; \
+   composer require typo3/cms-backend:^11.0.0; \
+   composer require typo3/cms-belog:^11.0.0; \
+   composer require typo3/cms-beuser:^11.0.0; \
+   composer require typo3/cms-core:^11.0.0; \
+   composer require typo3/cms-dashboard:^11.0.0; \
+   composer require typo3/cms-extbase:^11.0.0; \
+   composer require typo3/cms-extensionmanager:^11.0.0; \
+   composer require typo3/cms-felogin:^11.0.0; \
+   composer require typo3/cms-filelist:^11.0.0; \
+   composer require typo3/cms-fluid:^11.0.0; \
+   composer require typo3/cms-fluid-styled-content:^11.0.0; \
+   composer require typo3/cms-form:^11.0.0; \
+   composer require typo3/cms-frontend:^11.0.0; \
+   composer require typo3/cms-impexp:^11.0.0; \
+   composer require typo3/cms-info:^11.0.0; \
+   composer require typo3/cms-install:^11.0.0; \
+   composer require typo3/cms-recordlist:^11.0.0; \
+   composer require typo3/cms-rte-ckeditor:^11.0.0; \
+   composer require typo3/cms-seo:^11.0.0; \
+   composer require typo3/cms-setup:^11.0.0; \
+   composer require typo3/cms-sys-note:^11.0.0; \
+   composer require typo3/cms-t3editor:^11.0.0; \
+   composer require typo3/cms-tstemplate:^11.0.0; \
+   composer require typo3/cms-viewpage:^11.0.0; \
+    \
+   composer config platform.php 7.4.1; \
+   composer config sort-packages true; \
+   composer config extra.typo3/cms.web-dir public; \
+   \
+   composer normalize; \
```

Comment:
This won't work without https://packagist.org/packages/ergebnis/composer-normalize

Reply:
I removed that dependency.